src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheck.java

```diff
@@ -343,7 +343,7 @@ private void visitUnitaryOperator(DetailAST ast, int basicBranchingFactor) {
      * Leaves ternary operator (?:) and return tokens.
      */
     private void leaveUnitaryOperator() {
-        if (!afterValues.pop()) {
+        if (Boolean.FALSE.equals(afterValues.pop())) {
```

Comment:
What about `!afterValues.pop().booleanValue()`?

Reply:
Sonar do not like the fact that `afterValues.pop()` might be null, so your code still prone for NPE.

Reply:
Old code was prone to NPE as this is basically what it was converting the code to. If we had an issue with NPE here, regression would have found it.

Reply:
we do not have issue, it is the same over caution as https://checkstyle.org/config_coding.html#EqualsAvoidNull , .... if code is changed somewhere in wrong way at least this line will not be stable.